### PR TITLE
feat(forge): getLabel cheatcode

### PR DESCRIPTION
**Pull request by AlbertoCentonze**
## Motivation
As discussed in the foundry Telegram group, a `getLabel` cheatcode would be really useful when using `console.log` to access the label of an address without having to read it from the stack trace.

## Solution
The cheatcode was implemented follwing @mds1 guidance and the following [guide](https://dev.to/teddav/foundry-add-a-cheatcode-5hc8). However it's the first time I've written some rust in my whole life. I would be glad to receive some feedback on some not so idiomatic constructs I might have used.

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/util.rs
Comment (DaniPopes) [inline evm/src/executor/inspector/cheatcodes/util.rs:156]: 1. `Type::from(x)` -> `x.into()`, unless it can't be inferred
    - also `format!()` returns `String`, so this can be removed entirely

2. `option.or(Some(x)).unwrap()` -> `option.unwrap_or(x)` -> `option.unwrap_or_else(|| x)`
    - the last step is making it lazy since it's a function call, see [`Option::unwrap_or` docs](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.unwrap_or)

```suggestion
            let label = state.labels.get(&inner.0).cloned().unwrap_or_else(|| format!("unlabeled:{}", inner.0));
            Ok(ethers::abi::encode(&[Token::String(label)]).into())
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]
Thanks! I'm not sure about the return string when there is no label, since `format!("{}", address)` will truncate: `0x1234...4321`. This should probably be using the [Debug format](https://doc.rust-lang.org/stable/std/fmt/trait.Debug.html) ("{:?}"), but I'll defer to @mds1 @mattsse

**Review by mattsse — COMMENTED** [review]
lgtm, pending nits raised by @DaniPopes 

would appreciate a followup for
forge-std and book

**Comment by AlbertoCentonze** [comment]
Hey, thanks for your help. If you could point me towards some already existing tests for cheatcodes I would be glad to test the `getLabel` properly.

**Comment by mattsse** [comment]
nice,

for tests you can copy this file:

https://github.com/foundry-rs/foundry/blob/d76d9ae27b2cb8d6f79a6d0d815dc7f6ca802c6c/testdata/cheats/Label.t.sol#L7-L13

to a new file `GetLabel.sol` perhaps, same folder.

the existing cheats test will automatically pick it up and run it:
https://github.com/foundry-rs/foundry/blob/d76d9ae27b2cb8d6f79a6d0d815dc7f6ca802c6c/forge/tests/it/cheats.rs#L8-L10
so we only need a new sol test file

**Comment by Evalir** [comment]
Hey @AlbertoCentonze ! Thx for the contribution 😄 . I've added a few commits to rebase master, format things, and add a test. Thank you!

**Review by mattsse — APPROVED** [review]
gg

**Comment by AlbertoCentonze** [comment]
Amazing thanks!

**Pull request merged**
